lib/aurora_bootstrapper/exporter.rb

```diff
@@ -4,30 +4,43 @@ module AuroraBootstrapper
   class Exporter
     attr_reader :client
 
-    def initialize( client:, prefix: "", export_bucket:, blacklisted_tables: "", whitelisted_tables: "", blacklisted_fields: "" )
+    def initialize( client:, prefix: "", export_bucket:, blacklisted_tables: "", whitelisted_tables: "", blacklisted_fields: "", notifier: nil )
       @match              = "#{prefix}.*"
       @export_bucket      = export_bucket
       @blacklisted_tables = blacklisted_tables.split(",")
       @whitelisted_tables = whitelisted_tables.split(",")
       @blacklisted_fields = blacklisted_fields.split(",")
       @client             = client
+      @notifier           = notifier
+      d                   = DateTime.now
+      d_str               = d.strftime("%Y-%m-%d")
+      @export_date        = ENV.fetch( 'EXPORT_DATE', d_str )
     end
 
     def export!
       @client.query( "set sql_mode='NO_BACKSLASH_ESCAPES'" )
+
       database_names.all? do | database_name |
         begin
           @client.query( "use `#{database_name}`" )
           database = Database.new database_name: database_name,
                                          client: @client,
                              blacklisted_tables: @blacklisted_tables,
                              whitelisted_tables: @whitelisted_tables,
-                             blacklisted_fields: @blacklisted_fields
+                             blacklisted_fields: @blacklisted_fields,
+                             export_date: @export_date 
           database.export! into_bucket: @export_bucket
         rescue => e
           AuroraBootstrapper.logger.error message: "Error in database #{database_name}", error: e
         end
       end
+
+      result = true
+      if @notifier
+        result = @notifier.push_state?( export_date: @export_date, into_bucket: @export_bucket )
```

Comment:
since this is the only place we use export date and Notifier is the class that uses it, can it handle its derivation?
like

```ruby
class Exporter
  def notifier
    @notifier ||= Notifier.new( bucket: @export_bucket )
  end
  
  def export!
    #...
    notifier.notify.success?
  end
end

class Notifier
  def initialize( bucket: nil ); end
  
  def export_date
    @export_date ||= ENV.fetch( 'EXPORT_DATE',
                                 DateTime.now.strftime("%Y-%m-%d") )
  end
end
```

Reply:
but table.rb also uses export_date to construct the s3 path to put data files

Reply:
pass along @notifier.export_date all the way from exporter.rb to database.rb to table.rb